sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -325,6 +325,21 @@ class SessionCatalog(
     new Path(new Path(dbLocation), formatTableName(tableIdent.table)).toString
   }
 
+  // ----------------------------------------------
+  // | Methods that interact with temp views only |
+  // ----------------------------------------------
+
+  /**
+   * Return a temporary view exactly as it was stored.
+   */
+  def getTempView(name: String): Option[LogicalPlan] = synchronized {
```

Comment:
we already have `createTempView`, I don't think the new APIs make `SessionCatalog` more complicated.